Adding docs for leader election timings (#635)
elastic/ingest-docs: docs/en/ingest-management/elastic-agent/configuration/providers/kubernetes_leaderelection-provider.asciidoc

default QPS and burst settings are used.
`leader_lease`:: (Optional) Specify the name of the leader lease.
This is set to `elastic-agent-cluster-leader` by default.
`leader_retryperiod`:: (Optional) Default value 2 (in sec). How long before {agent}s try to get the `leader` role. 

> `leader_leaseduration`:: (Optional) Default value 15 (in sec).  How long the Leader {agent} holds the `leader` state.

Reviewer: ```suggestion
`leader_leaseduration`:: (Optional) Default value 15 (in sec).  How long the leader {agent} holds the `leader` state.
```

> `leader_retryperiod`:: (Optional) Default value 2 (in sec). How long before {agent}s try to get leader role. 
> `leader_leaseduration`:: (Optional) Default value 15 (in sec).  How long the Leader {agent} to hold the "leader" state
> `leader_renewdeadline`:: (Optional) Default value 10 (in sec). How long for leaders to retry getting "leader" role

Reviewer 2: ```suggestion
`leader_retryperiod`:: (Optional) Default value 2 (in sec). How long before {agent}s try to get the `leader` role. 
`leader_leaseduration`:: (Optional) Default value 15 (in sec).  How long the Leader {agent} holds the `leader` state.
`leader_renewdeadline`:: (Optional) Default value 10 (in sec). How long leaders retry getting the `leader` role.
```

|===

[discrete]
= Understanding leader timings

> As described above, the LeaderElection configuration offers following parameters: Lease duration (`leader_leaseduration`), Renew deadline (`leader_renewdeadline`) and  
> Retry period (`leader_retryperiod`) . Based on the config provided, each agent will trigger {k8s} API requests and will try to check the status of the lease. 

Reviewer 2: ```suggestion
As described above, the LeaderElection configuration offers the following parameters: Lease duration (`leader_leaseduration`), Renew deadline (`leader_renewdeadline`), and  
Retry period (`leader_retryperiod`). Based on the config provided, each agent will trigger {k8s} API requests and will try to check the status of the lease. 
```

> NOTE: The number of Leader API calls is proportional to the number of {agent}s installed. This means that Leader API requests will come from all {agent}s per `leader_retryperiod`. Setting `leader_retryperiod` to a greater value than the default (2sec), means that less API requests will be made towards {k8s} Control API, but will also increase the period where collection of metrics from Leader {agent} might be lost.

Reviewer 2: ```suggestion
NOTE: The number of Leader API calls is proportional to the number of {agent}s installed. This means that Leader API requests will come from all {agent}s per `leader_retryperiod`. Setting `leader_retryperiod` to a greater value than the default (2sec), means that fewer API requests will be made towards the {k8s} Control API, but will also increase the period where collection of metrics from the leader {agent} might be lost.
```

> NOTE: The number of Leader API calls is proportional to the number of {agent}s installed. This means that Leader API requests will come from all {agent}s per `leader_retryperiod`. Setting `leader_retryperiod` to a greater value than the default (2sec), means that fewer API requests will be made towards the {k8s} Control API, but will also increase the period where collection of metrics from the leader {agent} might be lost.

Reviewer: ```suggestion
NOTE: The number of leader calls to the K8s Control API is proportional to the number of {agent}s installed. This means that requests will come from all {agent}s per `leader_retryperiod`. Setting `leader_retryperiod` to a greater value than the default (2sec), means that fewer requests will be made towards the {k8s} Control API, but will also increase the period where collection of metrics from the leader {agent} might be lost.
```

The library applies https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L76[specific checks] for the timing parameters and if those are not verified the {agent} will exit with `panic` error.

In general: 
- Leaseduration must be greater than renewdeadline
- Renewdeadline must be greater than retryperiod*JitterFactor. 

> NOTE: Constant JitterFactor=1.2 defined in https://pkg.go.dev/gopkg.in/kubernetes/client-go.v11/tools/leaderelection[leaderelection lib]

Reviewer 2: ```suggestion
NOTE: Constant JitterFactor=1.2 is defined in https://pkg.go.dev/gopkg.in/kubernetes/client-go.v11/tools/leaderelection[leaderelection lib].
```
